packages/react-topology/README.md

```diff
@@ -29,11 +29,99 @@ or
+
+Use a controller to wrap your topology view component. In the example below, this is done via the `VisualizationProvider` which consumes the `Controller` via context.
+
+Three `register` methods are accessed by the controller.
+
+The following two must be declared explicitly\:
+
+- `registerLayoutFactory`\: This method sets the layout of your topology view (e.g. Force, Dagre, Cola, etc.). You can use `defaultLayoutFactory` as a parameter if your application supports all layouts. You can also update `defaultLayout` to a custom implementation if you only want to support a subset of the available layout options.
+
+- `registerComponentFactory`\: This method lets you customize the components in your topology view (e.g. nodes, groups, and edges). You can use `defaultComponentFactory` as a parameter.
+
+The register method below is initialized in `Visualization.ts`, therefore it doesn't need to be declared unless you want to support a custom implementation which modifies the types.
+
+- `registerElementFactory`\: This method sets the types of the elements being used (e.g. graphs, nodes, edges). `defaultElementFactory` uses types from `ModelKind` and is exported in `index.ts`.
+
+#### Example Component Usage
 
+```ts
+import * as React from 'react';
+import {
+  SELECTION_EVENT,
+  SelectionEventListener,
+  TopologyView,
+  useEventListener,
+  useVisualizationController,
+  Visualization,
+  VisualizationProvider,
+  VisualizationSurface
+} from '@patternfly/react-topology';
+import stylesComponentFactory from './components/stylesComponentFactory';
+import defaultLayoutFactory from './layouts/defaultLayoutFactory';
+import defaultComponentFactory from './components/defaultComponentFactory';
+import { generateDataModel } from './data/generator';
+import { useTopologyOptions } from './useTopologyOptions';
+
+const TopologyViewComponent: React.FunctionComponent = () => {
+  const [selectedIds, setSelectedIds] = React.useState<string[]>([]);
+  const controller = useVisualizationController();
+
+  const { nodeOptions, edgeOptions, nestedLevel, creationCounts } = useTopologyOptions(controller);
+
+  React.useEffect(() => {
+    const dataModel = generateDataModel(
+      creationCounts.numNodes,
+      creationCounts.numGroups,
+      creationCounts.numEdges,
+      nestedLevel,
+      nodeOptions,
+      edgeOptions
+    );
+
+    const model = {
+      graph: {
+        id: 'g1',
+        type: 'graph',
+        layout: 'Cola'
+      },
+      ...dataModel
+    };
+
+    controller.fromModel(model, false);
+    // Don't update on option changes, its handled differently to not re-layout
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [creationCounts]);
+
+  useEventListener<SelectionEventListener>(SELECTION_EVENT, ids => {
+    setSelectedIds(ids);
+  });
+
+  return (
+    <TopologyView>
+      <VisualizationSurface state={{ selectedIds }} />
```

Comment:
Haven't yet added documentation for the way the `TopologyViewComponent` is built here. What do you think would be a good way to briefly explain what is getting passed to `state`?

Reply:
The state is application specific. It can be any data the application want to be able to store/retrieve from the controller. Adding state to the surface allows hooks to update when state changes.  The state is useful to keep graph state such as selected elements.